pkg/portal/cluster/fetcher.go

```diff
@@ -41,13 +44,28 @@ type client struct {
 // contains Kubernetes clients and returns the frontend-suitable data
 // structures. The concrete implementation of FetchClient wraps this.
 type realFetcher struct {
-	log           *logrus.Entry
-	configCli     configclient.Interface
-	kubernetesCli kubernetes.Interface
-	machineClient machineclient.Interface
+	log              *logrus.Entry
+	configCli        configclient.Interface
+	kubernetesCli    kubernetes.Interface
+	machineClient    machineclient.Interface
+	azureSideFetcher azureSideFetcher
 }
 
-func newRealFetcher(log *logrus.Entry, dialer proxy.Dialer, doc *api.OpenShiftClusterDocument) (*realFetcher, error) {
+type azureSideFetcher struct {
+	resourceGroupName string
+	subscriptionDoc   *api.SubscriptionDocument
+	env               env.Interface
+}
```

Comment:
I am not sure I understand the point of this, it is not really a fetcher, and its fields could just be in the `realFetcher`

Reply:
So the realFetcher is something which has stuff used for fetching details about a cluster which is in cosmosDB. I wanted to open doors to fetch from Azure directly like VM status VMID etc and things required for that can land in this struct. If any other stuff in future requires to fetch anything from Azure which might require any other information for that, it can be added here.

Reply:
but then we will do it in a segregated manner which I think will help a new engineer to understand the code easily.

Reply:
this does not help right now, it's not even a fetcher so it's confusing. I stand by my comment

Reply:
so what design do you suggest here?

Reply:
I already suggested a way to do it

Reply:
to put the fields in the realfetcher itself .. I really think a segregation must be here to the required things to fetch from Azure(not from kube-apiserver).. maybe we can call it something else as you mentioned it is not a fetcher .. we can call it something else but I think the segregation must be there.

Reply:
actually it should be a completely separated client. we don't need any of the existing realfetcher underlying clients do we? (only log ) 
the real fetcher fetches stuff from kubernetes and it should keep doing just that. If we need to fetch from azure, we should create a new and separate client